jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -217,225 +206,174 @@ public void execute(Runnable task) throws RejectedExecutionException
+                            return;
+
+                        // shrink if we are already removed or there are other reserved threads.
+                        // There is a small chance multiple threads will shrink below minSize
+                        if (getAvailable() > _minSize && _threads.remove(this, slot))
                         {
-                            // We will become a reserved thread if we can update the count below.
-                            next = State.RESERVED;
-                            size++;
+                            if (LOG.isDebugEnabled())
+                                LOG.debug("{} reservedThread shrank {}", ReservedThreadExecutor.this, this);
+                            return;
                         }
+                        task = waitForTask();
                     }
 
-                    // Update count for pending and size
-                    if (!_count.compareAndSet(count, pending, size))
-                        continue;
-
-                    if (LOG.isDebugEnabled())
-                        LOG.debug("{} was={} next={} size={}+{} capacity={}", this, _state, next, pending, size, _capacity);
-                    _state = next;
-                    if (next != State.RESERVED)
-                        break;
-
-                    // We are reserved whilst we are waiting for an offered _task.
-                    Runnable task = reservedWait();
-
-                    // Is the task the STOP poison pill?
-                    if (task == STOP)
-                        break;
-
-                    // Run the task
                     try
                     {
-                        _state = State.RUNNING;
+                        if (LOG.isDebugEnabled())
+                            LOG.debug("{} reservedThread run {} on {}", ReservedThreadExecutor.this, task, this);
                         task.run();
                     }
-                    catch (Throwable e)
-                    {
-                        LOG.warn("Unable to run task", e);
-                    }
-                    finally
+                    catch (Throwable t)
                     {
-                        // Clear any interrupted status.
-                        Thread.interrupted();
+                        if (LOG.isDebugEnabled())
+                            LOG.debug("reserved error", t);
                     }
                 }
             }
-            finally
+            catch (Throwable t)
             {
                 if (LOG.isDebugEnabled())
-                    LOG.debug("{} exited {}", this, ReservedThreadExecutor.this);
-                _threads.remove(this);
+                    LOG.debug("reserved threw", t);
+            }
+            finally
+            {
                 _thread = null;
+                // Clear any interrupted status.
+                if (Thread.interrupted() && LOG.isDebugEnabled())
+                    LOG.debug("interrupted {}", this);
```

Comment:
The log is confusing as the thread has not been interrupted -- remove the log statement, this is just cleanup code, no need to log.

Reply:
It will only be logged if the thread has been interrupted, as otherwise `Thread.interrupted()` returns false.

Having the log is not hugely important but it does get rid of the warning that the return of  `Thread.interrupted()` is ignored.

Reply:
We don't need this, and it is confusing because it logs "interrupted Thread" when in fact the thread is now _not_ interrupted.
I'm still for removing it as it is unnecessary.

Reply:
To be removed.